transport-classes-io_uring/src/main/java/io/netty/channel/uring/IOUringIoOps.java

```diff
@@ -0,0 +1,354 @@
+        this.bufferAddress = bufferAddress;
+        this.length = length;
+        this.offset = offset;
+        // We encode the fd, op and udata into a long so we are able to use it later.
+        this.udata = UserData.encode(fd, opcode, data);
+    }
+
+    /**
+     * Returns the filedescriptor.
+     *
+     * @return  fd
+     */
+    public int fd() {
+        return fd;
+    }
+
+    /**
+     * Returns the opcode.
+     *
+     * @return  opcode
+     */
+    public byte opcode() {
+        return opcode;
+    }
+
+    /**
+     * Returns the flags that will be applied.
+     *
+     * @return  flags
+     */
+    public int flags() {
+        return flags;
+    }
+
+    /**
+     * Returns the priority.
+     *
+     * @return ioPrio
+     */
+    public short ioPrio() {
+        return ioPrio;
+    }
+
+    /**
+     * Returns the rwFlags that will be applied. These are specific to the opcode.
+     *
+     * @return  rwFlags
+     */
+    public int rwFlags() {
+        return rwFlags;
+    }
+
+    /**
+     * Returns the bufferAddress that will be used. This is specific to the opcode.
+     *
+     * @return  bufferAddress
+     */
+    public long bufferAddress() {
+        return bufferAddress;
+    }
+
+    /**
+     * Returns the length that will be used. This is specific to the opcode.
+     *
+     * @return  length
+     */
+    public int length() {
+        return length;
+    }
+
+    /**
+     * Returns the offset that will be used. This is specific to the opcode.
+     *
+     * @return  offset
+     */
+    public long offset() {
+        return offset;
+    }
+
+    /**
+     * Returns the data that the user attached to the op. This data will be passed back on completion.
+     *
+     * @return  data
+     */
+    public short data() {
+        return UserData.decodeData(udata);
+    }
+
+    /**
+     * Return the full udata that is a combination of {@link #fd()}, {@link #opcode()} and {@link #data()}.
+     *
+     * @return  udata.
+     */
+    public long udata() {
+        return udata;
+    }
+
+    @Override
+    public String toString() {
+        return "IOUringIoOps{" +
+                "opcode=" + opcode +
+                ", flags=" + flags +
+                ", ioPrio=" + ioPrio +
+                ", fd=" + fd +
+                ", rwFlags=" + rwFlags +
+                ", bufferAddress=" + bufferAddress +
+                ", length=" + length +
+                ", offset=" + offset +
+                ", udata=" + udata +
+                '}';
+    }
+
+    /**
+     * Returns a new {@code OP_ASYNC_CANCEL} {@link IOUringIoOps}.
+     *
+     * @param fd        the filedescriptor
+     * @param flags     the flags.
+     * @param udata     the {@link #udata()} of the previous {@link IOUringIoOps} that should be cancelled.
+     * @param data      the data
+     * @return          ops.
+     */
+    public static IOUringIoOps newAsyncCancel(int fd, int flags, long udata, short data) {
```

Comment:
We could basically add a method for any `OP_` that io_uring supports. I just focused on the ones that we need. At the end people can just construct the `IOUringIoOps` themselves. These methods just make things easier the same way as liburing makes things easier compared to use the raw io_uring interface